salt/modules/cmdmod.py

```diff
@@ -404,6 +404,19 @@ def _get_stripped(cmd):
 
         return win_runas(cmd, runas, password, cwd)
 
+    if runas and salt.utils.platform.is_darwin():
+        # we need to insert the user simulation into the command itself and not
+        # just run it from the environment on macOS as that
+        # method doesn't work properly when run as root for certain commands.
+        if isinstance(cmd, (list, tuple)):
+            cmd = ' '.join(cmd)
```

Comment:
This will break if any arguments contain spaces. You'll need to use shlex.quote from six which we've already imported into cmdmod.py as `_cmd_quote`. For example:

```python
cmd = ' '.join(map(_cmd_quote, cmd))
```

Note that you'll also need to add `map` to the `from salt.ext.six.moves import` in cmdmod.py to ensure that the itertools version is used in Python 2:

```python
from salt.ext.six.moves import range, zip, map
```